common/djangoapps/student/docs/decisions/0002-streak-celebration.md

```diff
@@ -0,0 +1,32 @@
+Streak Celebration
+---------------------
+
+Status
+------
+Accepted
+
+Context
+-------
+Adding a celebration for when a learner visits the learning mfe for 3 days in a row.
+
+
+Relevant Ticket:
+https://openedx.atlassian.net/browse/AA-304
+
+Decisions
+--------
+
+1. **Retrieving streak celebration data**
+We use the course home and courseware metadata APIs to retrieve the data determining whether we should celebrate a streak.
+Some MFEs implement progressive loading, where individual requests only retrieve the data necessary for a given component. However, the learning MFE has gone the direction of retrieving all the necessary data upfront.
+
+2. **No custom caching for streak celebration data**
+Although the streak celebration data cannot change more than once per day, it will be retrieved on every page load. This is because the streak celebration data comes through the course home and courseware metadata APIs. We considered adding custom caching for the streak celebration data, but decided against it in order to avoid overcomplicating the metadata APIs.
```

Comment:
These first two points don't really fit an ADR I don't think. They are just documenting existing decisions that your new code is following.

Number 1 is an interesting ADR by itself maybe, on the MFE side. Something about "The MFE prefers to optimize on number of calls, rather than allowing for progressive loading" - that is an interesting tradeoff, and one I'm not fully sure of the arguments on either side of. Something like progressive loading being slower overall, but possibly a nicer UX?

But that's an MFE wide ADR, not a bullet point on this feature, I would think?

And Number 2 is a follow-on of Number 1. If you're going to stick this data in the middle of an existing API, of course you're not going to do custom caching on it. You would want a more holistic approach, caching on either backend or frontend of the whole API call. Doesn't feel like a streak decision specifically.

Numbers 3 and 4 are fine. They feel small for an ADR, but I understand that different teams have different thresholds.

Reply:
i think Number 4 differs from the others in that it isn't really a technical decision so i've deleted this file and moved the context of #3 into the docstring of the function that's getting the user timezone for the streak